core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -3163,7 +3166,9 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Nod
                 return superType;
             }
 
-            throw semanticException(TYPE_MISMATCH, node, message, firstType, secondType);
+            @SuppressWarnings("FormatStringAnnotation") // Error Prone wants the types of format arguments to be the same as where @FormatString is declared, but we need them to be different
+            TrinoException exception = semanticException(TYPE_MISMATCH, node, message, firstType, secondType);
```

Comment:
new TrinoException?

Reply:
I don't know, I think it's valid, just hits some limitations of Java type system (or Error Prone's interpretation of it)